input-output-hk/lace
# feat(extension): LW-6779 manual discovery trigger in settings

# Checklist

- [x] JIRA - [\<link>](https://input-output.atlassian.net/browse/LW-6779)

---

## Proposed solution

A button in settings triggering a address discovery

## Testing

Scenario 1:
1. Install extension on two browsers/accounts
2. restore same wallet in both places
3. turn on the offline mode in the first installation (Network tab -> throttling)
4. make a delegation in the second installation to at least one more pool and wait till it finishes
5. turn off the offline mode in the first installation
6. go to settings
7. click sync.

Scenario 2:
1. Restore a wallet and click go to my wallet button fast enough
2. see the sync overlay

## Screenshots

Settings option:
<img width="981" alt="image" src="https://github.com/input-output-hk/lace/assets/9270461/4dd6c0c8-b713-48a9-832f-5cfde5792cf4">
Overlay:
<img width="981" alt="image" src="https://github.com/input-output-hk/lace/assets/9270461/5c97ff93-a68c-43d4-83fe-8248d1284dbe">
Success toast:
<img width="981" alt="image" src="https://github.com/input-output-hk/lace/assets/9270461/4f647093-5ab8-4c38-a2b4-38523c035645">
Error toast:
<img width="937" alt="image" src="https://github.com/input-output-hk/lace/assets/9270461/30fc7536-64c3-43f8-8793-29595d0763c8">




---
# Allure report
`allure-report-publisher` generated test report!



**smokeTests**: ❌ [test report](https://lace-qa:8PP5wBaDV6UoXj@dq4ajm5i5q7bz.cloudfront.net/linux/chrome/2781/6699826073/index.html) for [b4490b4c](https://github.com/input-output-hk/lace/pull/687/commits/b4490b4cf4afabb92998930af06428afb2afc37f)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 29     | 3      | 0       | 0     | 32    | ❌     |
